From: Ben Hutchings <ben@decadent.org.uk>
Date: Wed, 29 Nov 2017 02:31:04 +0000
Subject: Revert "phy: increase size of MII_BUS_ID_SIZE and bus_id"
Forwarded: not-needed

This reverts commit a88a90128888dba8754db5a194dba84e9703b93f, which
was commit 4567d686f5c6d955e57a3afa1741944c1e7f4033 upstream.  I don't
want to change ABI for this.

---
--- a/include/linux/phy.h
+++ b/include/linux/phy.h
@@ -142,7 +142,11 @@ static inline const char *phy_modes(phy_
 /* Used when trying to connect to a specific phy (mii bus id:phy device id) */
 #define PHY_ID_FMT "%s:%02x"
 
-#define MII_BUS_ID_SIZE	61
+/*
+ * Need to be a little smaller than phydev->dev.bus_id to leave room
+ * for the ":%02x"
+ */
+#define MII_BUS_ID_SIZE	(20 - 3)
 
 /* Or MII_ADDR_C45 into regnum for read/write on mii_bus to enable the 21 bit
    IEEE 802.3ae clause 45 addressing mode used by 10GIGE phy chips. */
@@ -598,7 +602,7 @@ struct phy_driver {
 /* A Structure for boards to register fixups with the PHY Lib */
 struct phy_fixup {
 	struct list_head list;
-	char bus_id[MII_BUS_ID_SIZE + 3];
+	char bus_id[20];
 	u32 phy_uid;
 	u32 phy_uid_mask;
 	int (*run)(struct phy_device *phydev);
